### ISSUE TITLE
Startup via another Add-In: Unhandled exception

### ISSUE THREAD

**Issue by makuepfer** [rubberduck-vba/Rubberduck#4934]
**Rubberduck version information**

Version 2.4.1.4627
OS: Microsoft Windows NT 10.0.17134.0, x64
Host Product: Microsoft Office x86
Host Version: 16.0.10730.20334
Host Executable: WINWORD.EXE

**Description**
RubberDuck throws an unhandled exception, when you open the VBEditor via a C# VSTO addin in Word.

**To Reproduce**
1. Make sure that all Word instances are completly closed.
2. Create a "Word 2013 and 2016 VSTO Add-In" in Visual Studio and add the following code snipped to the addin startup or use my [example project](https://github.com/makuepfer/VBE_Starter).

```csharp
Globals.ThisAddIn.Application.ShowVisualBasicEditor = true;
```

3. Run the Add-In

3. Word starts and the RubberDuck error message box appears.

**Expected behavior**
The VBEditor opens and RubberDuck is started correcly and without any errors.

**Screenshots**
The following exception is displayed:

![rubberduck_startup_exception](https://user-images.githubusercontent.com/28920033/56734551-6f67af00-6763-11e9-832b-09838bbc4870.png)

**Logfile**
No logs are written, even if log level is set to trace.

**Comment by retailcoder** [comment]
Thought of marking this bug as [critical] because interfering with other add-ins is a big no-no.. however if I understand correctly it's just us failing to load (needs investigation), and the VBE still ends up opening, just without RD loaded, correct?

**Comment by makuepfer** [comment]
Yes, VBE appears with all its standrad content but RD is not loaded. 
You have to close all the word instances and open the VBE by hand afterwards.

**Comment by retailcoder** [comment]
https://github.com/makuepfer/VBE_Starter/blob/254126d81a93c540057f49090f6cc3d771fcb624/Makro_Starter/ThisAddIn.cs#L12-L17

```csharp
    public partial class ThisAddIn
    {
        private void ThisAddIn_Startup(object sender, System.EventArgs e)
        {
            Globals.ThisAddIn.Application.ShowVisualBasicEditor = true;
        }
```
IIRC VSTO add-ins are loaded when the host application starts up, while the splash form is still being displayed. If this startup code runs before the VBE is ready to be created, Rubberduck might be choking on COMExceptions trying to access COM objects that don't exist yet. Note: Rubberduck has no problem loading without a host document.
We now know that the VBE loads its add-ins before it finishes creating its own object model: loading the VBE before the host itself is fully loaded... not sure we can fix this.

**Comment by makuepfer** [comment]
Got your point. But this application is just an example to show how I call the VBE.

Our use case is the following:
We've got an word-ribbon with a button to open up all the macro-files that are loaded in Word (Startup-Path of Word).
In the end, we open up the VBE. 
You can only click on that button, when Word has fully started up and a document is loaded.

It's not a big deal for us to leave away the line of code, that opens the VBE.
But I think it's pity that RubberDuck crashes when we want to open it automatically.
If you press alt + f11 it is able to show up.

**Comment by retailcoder** [comment]
Definitely needs investigation - it doesn't make sense that RD would crash if you bring up the VBE through VSTO but not with Alt+F11...

**Comment by bclothier** [comment]
Just to offer two more data points ---

As you may know already, different hosts has their own OM with different methods to activate Visual Basic Editor.

In Access, the command is `DoCmd.RunCommand acCmdVisualBasicEditor`

clicking a button that runs the code has no problems.

In Word, I've attached a docm file with a command button that does `Application.ShowVisualBasicEditor = True`. 

This does not cause error. However, bizarrely, I get a debug dialog saying the code execution has been interrupted even though there were no breakpoints or `Stop`.

[MCVE.zip](https://github.com/rubberduck-vba/Rubberduck/files/3118175/MCVE.zip)

**Comment by makuepfer** [comment]
I've updated my example project with a ribbon that contains a button to open VBE.
That's exactly how I can reproduce the issue, without interfering with the word startup sequence.

VBE gets called here: https://github.com/makuepfer/VBE_Starter/blob/ae08da93116355a4acc992165961c20433542d62/Makro_Starter/WordRibbon.cs#L18
